src/plugins/orangehrmBuzzPlugin/Api/BuzzLikeOnShareAPI.php

```diff
@@ -0,0 +1,173 @@
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA
+ */
+
+namespace OrangeHRM\Buzz\Api;
+
+use OrangeHRM\Buzz\Api\Model\BuzzLikeOnShareModel;
+use OrangeHRM\Buzz\Dto\BuzzLikeSearchFilterParams;
+use OrangeHRM\Buzz\Traits\Service\BuzzLikeServiceTrait;
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CollectionEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\Auth\AuthUserTrait;
+use OrangeHRM\Entity\BuzzLikeOnShare;
+use OrangeHRM\Entity\BuzzShare;
+
+class BuzzLikeOnShareAPI extends Endpoint implements CollectionEndpoint
+{
+    use AuthUserTrait;
+    use BuzzLikeServiceTrait;
+
+    public const FILTER_SHARE_ID = 'shareId';
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $buzzLikeSearchFilterParams = new BuzzLikeSearchFilterParams();
+        $buzzLikeSearchFilterParams->setShareId(
+            $this->getRequestParams()->getInt(
+                RequestParams::PARAM_TYPE_ATTRIBUTE,
+                self::FILTER_SHARE_ID
+            )
+        );
+
+        $this->setSortingAndPaginationParams($buzzLikeSearchFilterParams);
+
+
+        $likes = $this->getBuzzLikeService()->getBuzzLikeDao()->getBuzzLikeOnShareList($buzzLikeSearchFilterParams);
+        $likeCount = $this->getBuzzLikeService()->getBuzzLikeDao()->getBuzzLikeOnShareCount($buzzLikeSearchFilterParams);
+
+        return new EndpointCollectionResult(
+            BuzzLikeOnShareModel::class,
+            $likes,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => $likeCount])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                self::FILTER_SHARE_ID,
+                new Rule(Rules::POSITIVE),
+                new Rule(Rules::ENTITY_ID_EXISTS, [BuzzShare::class])
```

Comment:
remove this validation and. follow the below approach.
https://github.com/orangehrm/orangehrm/pull/1535/files#diff-059e542f32bb7cdac65aaa45dfad0a2bbed812979b77a58a333190cce849779aR159-R162

Reply:
added. https://github.com/orangehrm/orangehrm/pull/1537/commits/ddb33fcd4282ff7ce8a100571235a3876a2bf9e2#diff-7b1839d3b3930ac1cd5e7d6a8166de16dc894ab1525c11ce35b827a3cfc61f5dR58-R61